src/IIIFPresentation/API.Tests/Integration/ModifyRewrittenPathTests.cs

```diff
@@ -0,0 +1,418 @@
+using Core.Infrastructure;
+using Core.Response;
+using IIIF.Presentation.V3.Strings;
+using IIIF.Serialisation;
+using Microsoft.Extensions.DependencyInjection;
+using Models.API.Collection;
+using Models.API.Manifest;
+using Repository;
+using Test.Helpers.Helpers;
+using Test.Helpers.Integration;
+
+namespace API.Tests.Integration;
+
+[Trait("Category", "Integration")]
+[Collection(CollectionDefinitions.StorageCollection.CollectionName)]
+public class ModifyRewrittenPathTests : IClassFixture<PresentationAppFactory<Program>>
+{
+    private readonly HttpClient httpClient;
+    private readonly PresentationContext dbContext;
+    private readonly IETagManager etagManager;
+    private readonly string parent;
+    private const int Customer = 1;
+    
+    public ModifyRewrittenPathTests(StorageFixture storageFixture, PresentationAppFactory<Program> factory)
+    {
+        dbContext = storageFixture.DbFixture.DbContext;
+        
+        httpClient = factory.ConfigureBasicIntegrationTestHttpClient(storageFixture.DbFixture,
+            appFactory => appFactory.WithLocalStack(storageFixture.LocalStackFixture));
+        
+        etagManager = (IETagManager)factory.Services.GetRequiredService(typeof(IETagManager));
+        
+        parent = dbContext.Collections
+            .First(x => x.CustomerId == Customer && x.Hierarchy!.Any(h => h.Slug == string.Empty)).Id;
+
+        storageFixture.DbFixture.CleanUp();
+    }
+    
+    [Fact]
+    public async Task CreateCollection_CreatesCollection_WhenRewrittenPaths()
+    {
+        var slug = nameof(CreateCollection_CreatesCollection_WhenRewrittenPaths);
+        // Arrange
+        var collection = new PresentationCollection
+        {
+            Behavior =
+            [
+                Behavior.IsPublic,
+                Behavior.IsStorageCollection
+            ],
+            Label = new LanguageMap("en", ["test collection"]),
+            Slug = slug,
+            Parent = $"http://no-customer.com/collections/{RootCollection.Id}",
+        };
+
+        var requestMessage = HttpRequestMessageBuilder.GetPrivateRequest(HttpMethod.Post, $"{Customer}/collections",
+            collection.AsJson());
+        HttpRequestMessageBuilder.AddHostNoCustomerHeader(requestMessage);
+
+        // Act
+        var response = await httpClient.AsCustomer().SendAsync(requestMessage);
+        
+        var responseCollection = await response.ReadAsPresentationResponseAsync<PresentationCollection>();
+
+        var id = responseCollection!.Id!.Split('/', StringSplitOptions.TrimEntries).Last();
+
+        var fromDatabase = dbContext.Collections.First(c => c.Id == id);
+        var hierarchyFromDatabase = dbContext.Hierarchy.First(h => h.CustomerId == 1 && h.CollectionId == id);
+
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+        responseCollection.Parent.Should().Be($"http://no-customer.com/collections/{RootCollection.Id}");
```

Comment:
Use a shared var for this and `Parent` in payload to show that they should match?